packages/plugin-compiler/src/main/kotlin/io/realm/compiler/AccessorModifierIrGeneration.kt

```diff
@@ -376,17 +380,37 @@ class AccessorModifierIrGeneration(private val pluginContext: IrPluginContext) {
         val getter = property.getter
         val setter = property.setter
         getter?.apply {
+            /**
+             * Transform the getter to whether access the managed object or the backing field
+             * ```
+             * get() {
+             *      return this.`$realm$objectReference`?.let { it.getValue("propertyName") } ?: backingField
```

Comment:
While this is a really nice one-liner, we should probably at some point check exactly what is generated in JVM for this. Especially since this is our hottest path.

Reply:
I played around with various approaches and already inspected the IR dump and it seems quite reasonable to me: https://github.com/realm/realm-kotlin/blob/ct/reafactor-object-internal/packages/plugin-compiler/src/test/resources/sample/expected/00_ValidateIrBeforeLowering.ir#L117..L135